### PR TITLE
fix: 503 error fixed when navigating to homepage

### LINKED ISSUE #21915: [Bug]: 503 when navigating to homepage
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

In edit mode, navigating to homepage while the app is loading throws a 503.

- [ ] #21972

### Steps To Reproduce

https://www.loom.com/share/fc7a50f97a724227bdbccc63cf125da0

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Appsmith v1.9.13-SNAPSHOT

### PR DESCRIPTION
**Pull request by eco-monk**
## Description

TL;DR 503 error fixed when navigating to homepage while the editor is loading.

- `RESET_EDITOR_REQUEST` is an action that removes `currentApplication` from the `applicationsReducer`.
- When starting the editor, appEngine saga has an action `completeChore` which reads the `currentApplication` object.
- Now, when we go back to homepage (while the editor is still initializing)
  - `RESET_EDITOR_REQUEST` is dispatched (so `currentApplication` is cleared)
  - the appEngine still runs in the background and tries to read  `currentApplication` throwing an error.

Now we make sure appEngine is cancelled whenever `RESET_EDITOR_REQUEST` is triggered.

A follow up task to investigate how errors are thrown in app editor saga: #21972 

Fixes #21915

## Media
Before:
https://www.loom.com/share/eae1eeeec04f4b35a9d3be6ea8177b0a

After:
https://www.loom.com/share/09c328f81d8344569c0ff0b7a3d97713

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by hetunandu — APPROVED** [review]
LGTM

File: app/client/src/sagas/AppEngineSaga.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { generateAutoHeightLayoutTreeAction } from "actions/autoHeightActions";
+import type AppEngine from "entities/Engine";
+import { type AppEnginePayload, AppEngineApiError } from "entities/Engine";
+import AppEngineFactory from "entities/Engine/factory";
```
Comment (arunvjn) [inline app/client/src/sagas/AppEngineSaga.ts:4]: Is there any particular reason for refactoring this into a new file ? InitSagas.ts kind of lost its purpose with this refactor and most KT docs and videos start with initSagas.
Would recommend keeping this logic in initSagas itself.

**Comment by arunvjn** [comment]
Left a comment. Rest looks good!

File: app/client/src/sagas/AppEngineSaga.ts (hunk 1 above)
Comment (eco-monk) [reply app/client/src/sagas/AppEngineSaga.ts]: Thanks Arun, that makes sense.

**Comment by Richarex** [comment]
Tested this on the DP url looks good, 503 error is not coming when navigating to homepage while the editor is still loading.

**Review by hetunandu — APPROVED** [review]
